packages/suite/src/components/suite/modals/ReduxModal/DeviceContextModal/PassphraseWalletConfirmation.tsx

```diff
@@ -0,0 +1,61 @@
+import { TrezorDevice } from '@suite-common/suite-types';
+import { useState } from 'react';
+import { CardWithDevice } from 'src/views/suite/SwitchDevice/CardWithDevice';
+import { SwitchDeviceRenderer } from 'src/views/suite/SwitchDevice/SwitchDeviceRenderer';
+import { PassphraseWalletConfirmationStep1 } from './PassphraseWalletConfirmationStep1';
+import { PassphraseWalletConfirmationStep2 } from './PassphraseWalletConfirmationStep2';
+import { PassphraseWalletConfirmationStep3 } from './PassphraseWalletConfirmationStep3';
+import { ContentType } from './types';
+
+type PassphraseWalletConfirmationContentProps = {
+    onSubmit: (value: string, passphraseOnDevice?: boolean) => void;
+    onDeviceOffer: boolean;
+    onCancel: () => void;
+};
+
+type PassphraseWalletConfirmationProps = PassphraseWalletConfirmationContentProps & {
+    device: TrezorDevice;
+};
+
+const PassphraseWalletConfirmationContent = ({
+    onSubmit,
+    onDeviceOffer,
+    onCancel,
+}: PassphraseWalletConfirmationContentProps): React.JSX.Element | undefined => {
```

Comment:
If you add this `| undefined` to the return type, it wont help with type matching as `undefined` is the implicit return type

Reply:
Not sure why it is not working with `if`s but with switch it works

![image](https://github.com/trezor/trezor-suite/assets/152899911/a6af6da3-0994-4360-8f15-e65640eeb728)